htlcswitch/link.go

```diff
@@ -389,6 +389,11 @@ type channelLink struct {
 	// respect to the quiescence protocol.
 	quiescer quiescer
 
+	// stateQueries is a channel that is used to query the current state of
+	// the channelLink in a thread-safe manner by delegating the state reads
+	// to the main event loop.
+	stateQueries chan func()
```

Comment:
I think that we need to leave some warnings for the wise if we take this approach to link queries. 

Since we've got such an eclectic mix of locks and atomics, queries sent into this channel _still_ need to handle their own locking (or remove all non-main-loop access to the elements of the link they're dealing with).

As an example:
* `UpdateFwdPolicy` currently locks the link and sets new routing policy.
* If we were to add a policy-related query and use `stateQueries` it would either need to:
  * Move `UpdateFwdPolicy` to *also* go via `stateQueries` 
  * Include locking so that it's thread-safe with the out-of-main-loop `UpdateFwdPolicy`

I like the direction that this goes in, but I think the road there looks messy.